sdk/digitaltwins/Azure.DigitalTwins.Core/src/Models/BasicDigitalTwinComponent.cs

```diff
@@ -22,7 +22,7 @@ public class BasicDigitalTwinComponent
         /// The metadata property, required on a component to identify as one.
         /// </summary>
         [JsonPropertyName(DigitalTwinsJsonPropertyNames.DigitalTwinMetadata)]
-        public object Metadata { get; set; } = new object();
+        public BasicDigitalTwinComponentMetadata Metadata { get; set; } = new BasicDigitalTwinComponentMetadata();
```

Comment:
Rather than having a type here. I think we can do something like:

```
[JsonPropertyName(DigitalTwinsJsonPropertyNames.DigitalTwinMetadata)]
IDictionary<string, DigitalTwinPropertyMetadata> Metadata { get; set; } = new Dictinoary<string, DigitalTwinPropertyMetadata>();
```

This will allow us to strongly type the property metadata that exists on the components.

Reply:
I think that works. I'll give it a shot.

Reply:
Alternatively we can use a custom converter to avoid the limitations of JsonExtensionData.